### ISSUE TITLE
`plot3d` does not work with functions involving `Min`/`Max`

### ISSUE THREAD

**Issue by simeonkr** [sympy/sympy#25818]
For example:

```
x, y = symbols('x y')
plot3d(Min(x,y), (x, 0, 10), (y, 0, 10))
```

In v1.12, this produces the following error:
```
...
   1547 
   1548     def show(self):
-> 1549         self.process_series()
   1550         #TODO after fixing https://github.com/ipython/ipython/issues/1255
   1551         # you can uncomment the next line and remove the pyplot.show() call
...
---> 86     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     87 
     88 

TypeError: only integer scalar arrays can be converted to a scalar index```

**Comment by ArmanKukreti** [comment]
@simeonkr can you tell that the issue is present in which file path?

**Comment by ArmanKukreti** [comment]
Hi @simeonkr,

I've identified and fixed the issue related to plot3d does not work with functions involving Min/Max #25818. I'd like to contribute the code that resolves this issue.

Could you please let me know the file path where I should commit the code changes? I'm happy to create a pull request once I have the information.

Thanks!

**Comment by simeonkr** [comment]
Hi @ArmanKukreti. In my case the error occurs in [sympy/plotting/plot.py (v12)](https://github.com/sympy/sympy/blob/1.12/sympy/plotting/plot.py), but I suggest that you reproduce the bug to be sure.

**Comment by oscarbenjamin** [comment]
I think that this bug has already been fixed in the master branch since sympy 1.12. I tried this:
```python
In [1]: from sympy.plotting import plot3d

In [2]: x, y = symbols('x y')
   ...: plot3d(Min(x,y), (x, 0, 10), (y, 0, 10))
Out[2]: <sympy.plotting.backends.matplotlibbackend.matplotlib.MatplotlibBackend at 0x7feec9940440>
```
I got this figure:

![Figure_1](https://github.com/sympy/sympy/assets/1159732/2e59992f-407b-443a-8cc9-cc5c4b5e2cc0)
